src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ResourceDataNode.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Blob of data containing resources for all assemblies generated into the image.
+    /// Resources are simply copied from the inputs and concatenated into this blob.
+    /// All format information is provided by <see cref="ResourceIndexNode"/>
+    /// </summary>
+    internal class ResourceDataNode : ObjectNode, ISymbolNode
+    {
+        /// <summary>
+        /// Resource index information generated while extracting resources into the data blob
+        /// </summary>
+        private List<ResourceIndexData> IndexData { get; set; }
```

Comment:
Alas. Private properties are my workaround for not liking field prefixing. Made both private properties into fields.